aspnetcore-webapi/GettingStartedASPNETMongoDB/Test/StudentControllerUnitTest.cs

```diff
@@ -0,0 +1,131 @@
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using GettingStartedASPNETMongoDB.Controllers;
+using GettingStartedASPNETMongoDB.Interfaces;
+using GettingStartedASPNETMongoDB.Models;
+
+namespace Test
+{
+    public class StudentControllerUnitTest
+    {
+        const string id = "123456789054354";
+
+        private Mock<IStudentService> _studentService;
+
+        private Mock<ICourseService> _courseService;
+
+        private StudentsController _studentController;
+
+        private Student _student;
+
+        public StudentControllerUnitTest()
+        {            
+            SetUp();
+        }
+
+        [Fact]
+        public async Task GivenCreate_WhenSentCorrectData_ThenReturnCreatedStatus()
+        {  
+            //Act
+            var result = await _studentController.Create(_student);
+
+            //Assert
+            Assert.IsType<CreatedAtActionResult>(result);
+        }
+
+        [Fact]
+        public async Task GivenGetById_WhenSentCorrectId_ThenReturnStudentWithOkStatus()
+        {
+            //Act
+            var result = await _studentController.GetById(id);
+
+            //Assert
+            var okResult = Assert.IsType<OkObjectResult>(result.Result);
+
+            Assert.IsType<Student>(okResult.Value);            
+        }
+
+        [Fact]
+        public async Task GivenGetAll_WhenSentCorrectId_ThenReturnStudentsWithOKStatus()
+        {  
+            //Arrange
+            var students = new List<Student> 
+            { 
+                new Student 
+                { 
+                    Id = "1", 
+                    FirstName = "John", 
+                    LastName = "Doe" 
+                }, 
+                new Student 
+                { 
+                    Id = "2", 
+                    FirstName = "Jane", 
+                    LastName = "Doe" 
+                } 
+            };            
+
+            _studentService.Setup(x => x.GetAll()).ReturnsAsync(students);
+
+            _studentController = new StudentsController(_studentService.Object, _courseService.Object);
+
+            //Act
+            var result = await _studentController.GetAll();
+
+            //Assert
+            var okResult = Assert.IsType<OkObjectResult>(result.Result);
+
+            var model = Assert.IsAssignableFrom<IEnumerable<Student>>(okResult.Value);
+
+            Assert.Equal(students.Count, model.Count());           
```

Comment:
For consistency of the test, we should also verify that the returned students collection is the expected collection, i.e. that the students returned match what is expected.